src/utils/utf8.c

```diff
@@ -170,11 +196,18 @@ utf8_normalize(utf8_ctx_t *ctx, strbuf_t *buf)
 	}
 	len = unorm2_normalize(ctx->normalizer,
 	    buf1, -1 /* NUL-terminated */, buf2, __arraycount(buf2), &ec);
+
+	
 	if (__predict_false(U_FAILURE(ec) || (size_t)len == __arraycount(buf2))) {
 		return -1;
 	}
 	if (strbuf_prealloc(buf, len + 1) == -1) {
 		return -1;
 	}
+	int32_t textLim = (int32_t)len;
+	int32_t textLen = (int32_t)len;
+	utrans_transUChars(ctx->transliterator, buf2, &textLen, 100, 0, &textLim, &ec);
```

Comment:
- I think it it is better to introduce a new function `utf8_subs_diacritics()`. We might want to let the users configure whether they want this substitution or not. The good default, however, would be to enable it.
- While perfectly fine in C99, the coding convention we use is to declare the variables at the top of the function.